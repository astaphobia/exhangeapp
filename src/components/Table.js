import React from "react"
import PropTypes from "prop-types"
import get from "lodash/get"
import first from "lodash/first"

const Header = ({ children, width }) => (
  <th width={get(width, "")}>{children}</th>
)

class Table extends React.Component {
  static propTypes = {
    children: PropTypes.array,
  }
  static defaultProps = {
    children: [],
  }
  static Header = Header
  render() {
    return (
      <table
        className="table table-hover table-striped table-bordered"
        style={{ tableLayout: "fixed" }}
      >
        <thead className="thead-dark">
          <tr>{first(this.props.children)}</tr>
        </thead>
        <tbody>
          {React.Children.map(this.props.children, (child, index) => (
            <tr key={index}>
              {React.Children.map(child.props.children, (child, index) => {
                if (get(child.props, "renderColumn", false)) {
                  return (
                    <td key={index}>
                      {child.props.renderColumn(get(child.props, "row", "-"))}
                    </td>
                  )
                }
                return <td key={index}>{get(child.props, "row", "-")}</td>
              })}
            </tr>
          ))}
        </tbody>
      </table>
    )
  }
}

export default Table
