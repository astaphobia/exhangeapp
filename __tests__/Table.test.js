import React from 'react'
import { shallow } from 'enzyme'

import Table from '@/components/Table'

describe("Table", () => {
    it("should render perfectly", () => {
        const items = { one: 1, two: 2 }
        const comp = shallow(
            <Table>
                {Object.keys(items).map((key, index) => (
                    <React.Fragment key={index}>
                        <Table.Header row={key}>
                            Mock
                    </Table.Header>
                        <Table.Header row={key}>
                            Up
                    </Table.Header>
                    </React.Fragment>))}
            </Table>
        )
        expect(comp).toMatchSnapshot()
    })
})