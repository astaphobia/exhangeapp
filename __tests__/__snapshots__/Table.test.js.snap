// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render perfectly 1`] = `
<table
  className="table table-hover table-striped table-bordered"
  style={
    Object {
      "tableLayout": "fixed",
    }
  }
>
  <thead
    className="thead-dark"
  >
    <tr>
      <Header
        row="one"
      >
        Mock
      </Header>
      <Header
        row="one"
      >
        Up
      </Header>
    </tr>
  </thead>
  <tbody>
    <tr
      key=".$0"
    >
      <td
        key="0/.0"
      >
        one
      </td>
      <td
        key="1/.1"
      >
        one
      </td>
    </tr>
    <tr
      key=".$1"
    >
      <td
        key="0/.0"
      >
        two
      </td>
      <td
        key="1/.1"
      >
        two
      </td>
    </tr>
  </tbody>
</table>
`;
